src/App.tsx

```diff
@@ -1,14 +1,74 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodoId, setSelectedTodoId] = useState<number>(0);
+  const [filter, setFilter] = useState('all');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos);
+  }, []);
+
+  const selectFilter = (value: string) => {
+    setFilter(value);
+  };
+
+  const setFilterQuery = (value: string) => {
+    setQuery(value);
+  };
+
+  const selectTodoId = (todoId: number) => {
+    setSelectedTodoId(todoId);
+  };
+
+  const closeTodoModal = () => {
+    setSelectedTodoId(0);
+  };
```

Comment:
Wrap all callbacks you are passing to another components with the `useCallback`

Reply:
done